src/Models/Services/DtdlUtils.ts

```diff
@@ -516,6 +521,95 @@ export const isValidDtmiPath = (path: string, isFinal: boolean): boolean => {
 
 // #endregion
 
+type IValidationArgs<T extends DtdlInterface | DtdlInterfaceContent> = {
+    updatedItem: T;
+    originalItem: T;
+    selectedModelId: string;
+    existingModels: DtdlInterface[];
+};
+type IValidationResult =
+    | {
+          isValid: true;
+          updatedModel: DtdlInterface;
+      }
+    | {
+          isValid: false;
+          error: {
+              title: string;
+              message: string;
+          };
+      };
+export async function validateItemChangeBeforeSaving<
+    T extends DtdlInterface | DtdlInterfaceContent
+>(args: {
+    content: string;
+    originalItem: T;
+    selectedModelId: string;
+    existingModels: DtdlInterface[];
+}): Promise<IValidationResult> {
+    const isJsonStringValid = (value: string): T | null => {
+        try {
+            return JSON.parse(value);
+        } catch (e) {
+            return null;
```

Comment:
Should we return an object here with a message that the json didn't parse?

Reply:
I actually stole this block from the vendors, but basically, they null check the result to know if it parsed correctly. I could make it fancier but that is what they do in a couple places to avoid bubbling up errors from the parsing operation.